### DISCUSSION TITLE
CaptumExplainer in Explqin do not support ShapleyValueSamplin ?

### DISCUSSION THREAD

**Discussion by arthurserres** [pyg-team/pytorch_geometric#8625] [General]
Hello, I am using Captumexplainer in order to explain prediction of my GNN. To be more precise, the configuration that I use is 

`  explainer = Explainer(
        model=model,
        explanation_type='model',
        algorithm=CaptumExplainer('ShapleyValueSampling'),
        node_mask_type='attributes',
        edge_mask_type='object',
        model_config=model_config,
        threshold_config=dict(threshold_type  = 'topk', value=topk)
    )`

For some reason, computation of the explanation object is never ending although the graph is not that big... Is there a reason why I an stuck with this ? Maybe it is computationally hard, or simply this configuration does not make sense (not supported)? 

Thanks in advance for the clarification :)

**Comment by rachitk** [comment]
ShapleyValueSampling is very slow, and on larger graphs or larger models, it can take a lot of time. Generally, this should work, but in practice it can simply just be time-consuming.

You can try other methods such as `IntegratedGradients` if the Shapley option is computationally intractable.

**Reply by arthurserres** [reply]
Hi ! 

Thanks a lot for this super fast answer, this is really helpful! Indeed, IntegratedGradients is quite fast but I do not manage to understand how the contributions are computed for the edges. I made a post about this [here.](https://github.com/pyg-team/pytorch_geometric/discussions/8612).

Very good to know that Shapley values approach should work,I'll just have to be patient.

Thanks :)
